apps/service_providers/llm_service/history_managers.py

```diff
@@ -223,7 +228,7 @@ def add_messages_to_history(
         self, input: str, input_message_metadata: dict, output: str, output_message_metadata: dict, *args, **kwargs
     ):
         self.input_message_metadata = input_message_metadata
-        self.output_message_metadata = output_message_metadata
+        self.output_message_metadata = self.output_message_metadata | output_message_metadata
```

Comment:
This won't merge the lists. It will just overwrite so if 2 nodes try to attach files, only the last node will succeed